### DISCUSSION TITLE
Auto-Eat and Auto-Drink should use different methods to eat/drink

### DISCUSSION THREAD

**Discussion by Unrepentant-Atheist** [cataclysmbn/Cataclysm-BN#2801] [General]
### Is your feature request related to a problem? Please describe.

I'm not really seeing any system behind auto-eat and auto-drink except maybe last items dropped in said tile or calorie amount??

Eat nothing but acorns a season, and get scurvy. Put crackling, meat, and rose petals on tile and observe. Scurvy barely gets removed because PC keeps eating meat / cracklings even though PC has scurvy and needs vitamin C (rose petals).

### Describe the solution you'd like

It should prioritize as follows:

1. Eat/drink items that spoil soon
2. But keep in mind your vitamin a/b/c/calcium needs
3. Eat/drink everything else after that

Essentially vitamin needs > spoilage  > everything else . Vitamin needs beat spoilage timer in my opinion. Food is easy enough to forage, but that's a design decision for the devs.

**Comment by chaosvolt** [comment]
Vitamins are a hot mess as usual. Spoilage-first at least will be implemented in: https://github.com/cataclysmbnteam/Cataclysm-BN/pull/2402

Fun bonus fact: the current behavior of what autoeat picks first is pretty wacky, hence Viss having pestered me to change it for him. It's basically dependent on whichever item was dropped onto that tile first. If you look at a stack of items using the `/` AIM menu, then hit `s` to sort, selecting "unsorted" will show the same order.
